### PR TITLE
(Review needed for ARM manifest changes. will merge later )Migration changes dev mysql microsoft.d bfor my sql 2022 06 01 privatepreview

### PR DESCRIPTION
**Pull request by pranavi-dandu**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by NetRock — APPROVED** [review]

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1440]: >../../../../../common-types/resource-management/v1/types.json [](http://example.com/codeflow?start=21&length=61)

Looks like extraneous path stuff here.

Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1440]: >v1 [](http://example.com/codeflow?start=130&length=2)

Recommend using v3 common types.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1474]: >cutoverMigration [](http://example.com/codeflow?start=132&length=16)

POST actions should have verb names. Any reason this action isn't just named "migrate"?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2724]: >sourceServerResourceId [](http://example.com/codeflow?start=9&length=22)

The resource should be in the URI path rather than the body. POST actions on single resources should just be the resource path + the action name. In this case you wouldn't have a request payload.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2734]: >sourceServerResourceId [](http://example.com/codeflow?start=9&length=22)

If you move the POST action to the resource, you wouldn't need this property in the response either. I see you are reusing this definition for replicas below, so removing this property might require some refactoring.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2738]: >targetPlatformProviderId [](http://example.com/codeflow?start=9&length=24)

What is returned in this property? Does it have a fixed set of values?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2742]: >targetServerDetails [](http://example.com/codeflow?start=9&length=19)

I'm finding source and target to be a getting a little non-intuitive for these properties in the face of reports containing replica reports. So a report resource has a source server, a target server and an array of source replicas. Each source replica also has a source server, target server and an array of source replicas. What's in the replica array of a replica report?

Assuming I understand your API correctly, I think it would be a little clearer if you have a single machine assessment report definition and an outer assessment report definition that uses the other definition to lay out a primary and array of replicas.

Something more like:

machineAssessment: {
  serverResourceId,
  migrationInformation
}

assessmentReport: {
  primary: machineAssessment
  replicas: [
    machineAssessment,
    machineAssessment,
   ...
  ]
}

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2784]: >serverParameters [](http://example.com/codeflow?start=9&length=16)

serverConfiguration, or just configuration?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2779]: >serverResource [](http://example.com/codeflow?start=9&length=14)

If this is in ARM resourceId format, I'd recommend serverResourceId

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2814]: >severity [](http://example.com/codeflow?start=9&length=8)

Presumably this has a fixed set of values. If so, it should be a string enum.

**Comment by mentat9** [comment]
@pranavi-dandu - @NetRock asked me to also review your PR, so I've taken some time to look at it and provide comments, even though it's already approved by ARM.

Note, the first issue is that new API version PRs should be constructed using this process:
1. Copy all files from current API version to new API version folder.
2. Make a base commit.
3. Apply changes needed for new API version in subsequent commits (including updating the API version in each file).

If you use OpenAPIHub to generate your PR, it will automatically use the correct format.

This really helps reviewers see what's being changed in the new API version. Most ARM reviewers will require you resubmit your new API version PR in the correct format.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (pranavi-dandu) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1474]: Migrate would mean "start data migration".
Here cutoverMigration means complete the migration that was started

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (pranavi-dandu) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2734]: > @pranavi-dandu - @NetRock asked me to also review your PR, so I've taken some time to look at it and provide comments, even though it's already approved by ARM.
> 
> Note, the first issue is that new API version PRs should be constructed using this process:
> 
> 1. Copy all files from current API version to new API version folder.
> 2. Make a base commit.
> 3. Apply changes needed for new API version in subsequent commits (including updating the API version in each file).
> 
> If you use OpenAPIHub to generate your PR, it will automatically use the correct format.
> 
> This really helps reviewers see what's being changed in the new API version. Most ARM reviewers will require you resubmit your new API version PR in the correct format.

Yes agree with you. I created this PR via openapihub only

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (pranavi-dandu) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2779]: The ServerResource will hold all the properties of a server
The object will look like the one in this file : specification\mysql\resource-manager\Microsoft.DBforMySQL\preview\2022-06-01-privatepreview\examples\ServerGet.json

What is a arm resource Id model?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (pranavi-dandu) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2784]: ServerConfiguration and ServerParameters means the same thing. Wanted the name to be more user friendly and readable so named it as ServerParameters.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (pranavi-dandu) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2738]: The provider id and servertype is returned -> "/Microsoft.DBforMySQL/flexibleServers"
It is expected that the customers send "/Microsoft.DBforMySQL/flexibleServers", but incase they do not, we throw an exception

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (pranavi-dandu) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2724]: Here the API is hosted on Flexible Server RP.
And the sourceServerResourceId represent a single server resource Id.

Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2724]: OK, but I think this action should be performed on a specific flexible server (the same URI path that cutoverMigration is on). Then you don't have sourceServerResourceId in the body because it's already in the URI.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2738]: If it's a constant string, why doesn't your backend just fill it in? Why is it passed back and forth to/from the client as a separate property that never changes?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2779]: Right, missed that. Actually, the RPC doesn't allow embedding a resource in another resource. I'm trying to get a reading on whether that also applies to POST responses. In any case, this object is already called targetServerDetails, so I think "resource" would actually be a clearer name for this property.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2784]: That seems more confusing than friendly: the resource type is named "configurations" in the API. And seeing this serverParameters property, I'd also expect firewallRules to be named serverFirewallRules, for example. Since object is named "targetServerDetails", I think it sets the context as a server already. This approach seems clearer to me:

"targetServerDetails": {
  "resource": { ... }
  "configuration": { ... }
  "firewallRules": [ ... ]
}

WDYT?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2800]: >platformProviderId [](http://example.com/codeflow?start=9&length=18)

Is this also a string constant == "Microsoft.DBforMySQL/flexibleServers"? If so, I'd recommend removing it unless there's a scenario for it that I'm not getting.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1474]: OK. So this PR only adds two POST actions: assessForMigration and cutoverMigration. Presumably assessForMigration isn't starting a migration, so how does a client start a migration?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (NetRock) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1440]: do this for the entire file.

**Review by NetRock — COMMENTED** [review]
use v3 for entire file

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json]: The ARM guidance is that embedding a resource in a resource is not allowed in the resource model (GET/PUT/PATCH/DELETE). Instead, include a property that contains the resource ID of the other resource. For POST, that guidance is recommend but not required if there is a good reason to embed the resource. It's critical to ensure that there are no leaks across RBAC though. Can you ensure that anyone using this POST API already has RBAC read access to the server resource? If so, and you think it's important to have the full resource in this payload instead of the resource ID of the server, it's OK to leave this as an embedded resource.

**Comment by pranavi-dandu** [comment]
> @pranavi-dandu - I took another look and updated one of my comments, but there are still unaddressed questions. Can you address the remaining questions?

@mentat9 I replied to all of the comments and waiting for you to resolve them.

**Comment by mentat9** [comment]
> > @pranavi-dandu - I took another look and updated one of my comments, but there are still unaddressed questions. Can you address the remaining questions?
> 
> @mentat9 I replied to all of the comments and waiting for you to resolve them.

@pranavi-dandu - I resolved all of the ones you answered or changed. There are still 7 unaddressed/active comments here. Not sure why they aren't showing for you. You could also try using Codeflow if you don't see them above on this page...?

**Comment by pranavi-dandu** [comment]
> Codeflow

I still see pe

> > > @pranavi-dandu - I took another look and updated one of my comments, but there are still unaddressed questions. Can you address the remaining questions?
> > 
> > 
> > @mentat9 I replied to all of the comments and waiting for you to resolve them.
> 
> @pranavi-dandu - I resolved all of the ones you answered or changed. There are still 7 unaddressed/active comments here. Not sure why they aren't showing for you. You could also try using Codeflow if you don't see them above on this page...?

@mentat9  I dont see any new resolved ones from your side. For eg below are a few of the unresolved ones
image
image
image

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1474]: hi

**Comment by pranavi-dandu** [comment]
> > Codeflow
> 
> I still see pe
> 
> > > > @pranavi-dandu - I took another look and updated one of my comments, but there are still unaddressed questions. Can you address the remaining questions?
> > > 
> > > 
> > > @mentat9 I replied to all of the comments and waiting for you to resolve them.
> > 
> > 
> > @pranavi-dandu - I resolved all of the ones you answered or changed. There are still 7 unaddressed/active comments here. Not sure why they aren't showing for you. You could also try using Codeflow if you don't see them above on this page...?
> 
> @mentat9 I dont see any new resolved ones from your side. For eg below are a few of the unresolved ones image image image

Hey @mentat9 I don't know what the issue is, but whenever I reply to the comments, they don't show up on your screen or my colleagues. Only Im able to see my replies. But nonetheless, Im able to quote messages. This functionality is working fine. Did you ever see this issue?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1474]: This comment and all following comments are on the behalf of @pranavi-dandu 

The start migration is part of the server create call. We added a new enum string in CreateMode property for it. When the create mode is "Migrate", the server creation and data migration take place

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2724]: At the time of the assessment, the target server is not present.
The user first calls the assessForMigration api for their single server to check if the migration to a target flexible server would be successful or if there are any blockers.

And the reason for us to host the API on Target(Flexible) RP and not the Source(Single) RP is as below:
1.Single Server code base is a dying code base. In some months from now, we will not allow any more customers to create a single server
2.The code base is complex and hard to test any new changes right now
3.The train deployment takes months to roll out , and making any iterative changes would mean a long process for us to go to public preview and GA

And if the question is that, how are we getting the information about a single server from a Flexible RP,
Thankfully, we already have CAS endpoints(Internal apis) present on Single Server RP, and cross platform communication also available.
We just call the Single Server CAS endpoint from FLexible RP and get the information we need.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2734]: I will still need the SourceServerResourceID. It is not only used to represent primary but also the replica servers
Eg Response:
{
SourceServerResourceId : "......." (Primary source server resource Id )
ReplicaAssessmentReport : [
{
SourceServerResourceId: "......." (Replica source server resource Id 1)
}
{
SourceServerResourceId: "......." (Replica source server resource Id 2)
}
]
}

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2738]: I got that wrong. It is not sent by the customer in the request body. We send it as part of the response. It represents what is the target server platform the customer is looking to migrate to

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2742]: For your first question ie;
1.Each source replica also has a source server(represents the replica server), target server(target server for replica) and an array of source replicas. What's in the replica array of a replica report?
And: The replica array will remain empty for the source replica part.

2.My thoughts based on the new model you suggested
Here the primary can be the replica as well. By this what I mean is, the customer can call migration assessment directly on the replica as well.
In this case the model we defined makes more sense. WDYT?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json]: Changed it to
"targetServerDetails": {
"resource": { ... }
"configurations": { ... }
"firewallRules": [ ... ]
}

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json]: Here the ServerResource( which is a target server) in response is not yet created. As part of the assessment, we are just recommending a target server with set of configurations based on the source server.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (cpj2195) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2800]: this is a string constant. It is an identifier for which target platform the assessment is being performed for.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1474]: OK, I thought that could be the case. I see flexible servers don't have provisioningState today, which would help with ARM consistency in this scenario. Presumably, there is some state variable that indicates the migration has started and later completed, but the CreateMode approach seems to mix request and state a little bit. Given the history and that this is an incremental change, the only additional feedback I have on this API is that I'd recommend using completeMigration as the POST action to indicate it only does something if a migration is in progress.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2724]: I see: you specifically don't want the request routed on the server being migrated, so you put it at tenant level and pass in its ID. Given the situation, I think it makes sense.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2734]: OK, so the assessment of the source server also implicitly does assessment for all of the replicas of the source server. I found that part non-intuitive, since those are not really the source server (from the request). I think naming that property serverResourceId would actually be clearer.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2738]: OK, but there's only one possibility, right? I still don't see why the server needs to tell the client that the migration will be to a flexible server, since that's what they already asked about. Do you anticipate there will be other options in the future? If so, I think this should be a string enum with one value, maybe something like migrateTo: { mySqlFlexibleServer }. As you add more kinds of results, the enum would get additional values. If I've misunderstood, maybe you can clarify what you expect the client to do with this value.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2742]: Taking all comments into perspective, I think it works well if you change the name "primary" to "server" in the above structure. This also avoids creating a recursive definition. Either approach returns an empty replica array when the request is performed on a replica.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json]: Ah, so this is not a resource, but is constructed. Given that, it seems like this should just be the property bag. You don't have anything useful to fill in for systemData, id, tags, etc. right? Would it be better to call it properties and just include the properties?

**Comment by mentat9** [comment]
@pranavi-dandu, @cpj2195 - Thanks for powering through the comment issues. @pranavi-dandu I can see the images, but it's too hard to use them in responses, so I've responded to comment threads from @cpj2195. Also, quote comments seem to have reached some limit and aren't working now either. I vaguely recall seeing something like this once before, but it would have been a long time back: nothing recent.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json]: Sure, that's the usage model I visualized. I was suggesting that the serverResource property should contain the entire complex object, but without the ARM resource envelope (top-level properties like name, type, id, location, systemData, tags (we usually call it the property bag, since it's the contents of an object named "properties")). Those top-level properties need to be populated later at create time, not at assessment time. Consequently, if they are in the assessment payload, any values they have are meaningless. Does that make sense?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:2738]: I still find the source and target terms pretty confusing. My suggestion is to use the structure rather than source/target to clarify the meaning, but I won't block on this.

**Comment by mentat9** [comment]
Signing off with comments from ARM.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json
Comment (mentat9) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json:1474]: The reason cutoverMigration is less clear is because it implies you could do the whole job with that one action. It doesn't give any clue that some other action has to be done first. Using completeMigration for the name, clearly indicates that a start action has to be done first. This is my last comment in this thread: I won't block ARM approval on changing the name: you can decide.

**Comment by ArcturusZhang** [comment]
Hi @pranavi-dandu your PR is approved but the api-readiness check did not pass.
I added the `OkToMerge` label which works as merged if you want to update ARM's manifest.
Once your service is deployed, please ping me, and I will merge your PR

**Comment by pranavi-dandu** [comment]
> Hi @pranavi-dandu your PR is approved but the api-readiness check did not pass. I added the `OkToMerge` label which works as merged if you want to update ARM's manifest. Once your service is deployed, please ping me, and I will merge your PR

Sure I will let you know. As of now I do not want to merge the PR right away.

**Comment by ArcturusZhang** [comment]
> Hi @pranavi-dandu, one or multiple breaking change(s) is detected in your PR. Please check out the breaking change(s), and provide business justification in the PR comment and @ PR assignee why you must have these change(s), and how external customer impact can be mitigated. Please ensure to follow [breaking change policy](https://aka.ms/AzBreakingChangesPolicy) to request breaking change review and approval before proceeding swagger PR review. **Action**: To initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes). If you want to know the production traffic statistic, please see [ARM Traffic statistic](https://portal.azure-devex-tools.com/api/pr/query-traffic?pr=Azure/azure-rest-api-specs/19140&days=5). If you think it is false positive breaking change, please provide the reasons in the PR comment, report to Swagger Tooling Team via https://aka.ms/swaggerfeedback. **Note**: To avoid breaking change, you can refer to [Shift Left Solution](https://aka.ms/shiftleft) for detecting breaking change in early phase at your service code repository.

Hey @pranavi-dandu I rerun the pipelines and find that we now have cross-version breaking changes, this PR now requires the approval of breaking change review board. Please follow the guidance above to get that

**Pull request closed without merging**
